ogr/parsing.py

```diff
@@ -125,6 +117,46 @@ def parse_git_repo(potential_url: str) -> Optional[RepoUrl]:
         path = path[: -len(".git")]
 
     split = path.split("/")
+
+    return path, split
+
+
+def parse_path(
+    split: List[str], username: Optional[str]
+) -> Tuple[List[str], bool, Optional[str]]:
+    is_pagure_fork = "fork" == split[0] and len(split) >= 3
+    if is_pagure_fork:
+        # fork/username/namespace/repo format
+        username = split[1]
+        namespace_parts = split[2:-1]
+    else:
+        namespace_parts = split[:-1]
+
+    return namespace_parts, is_pagure_fork, username
+
+
+def parse_git_repo(potential_url: str) -> Optional[RepoUrl]:
+    """Cover the following variety of URL forms for Github/Gitlab repo referencing.
+
+    1) www.domain.com/foo/bar
+    2) (same as above, but with ".git" in the end)
+    3) (same as the two above, but without "www.")
+    # all of the three above, but starting with "http://", "https://", "git://", "git+https://"
+    4) git@domain.com:foo/bar
+    5) (same as above, but with ".git" in the end)
+    6) (same as the two above but with "ssh://" in front or with "git+ssh" instead of "git")
+    7) pagure format of forks (e.g. domain.com/fork/username/namespace/project
+
+    Notably, the repo *must* have exactly username and reponame, nothing else and nothing
+    more. E.g. `github.com/<username>/<reponame>/<something>` is *not* recognized.
```

Comment:
This is probably some old github-only note. Can you please update that so it matches the current state? (Change the wording to say that we don't support only main project URLs.)